include/rtlog/rtlog.h

```diff
@@ -243,7 +257,42 @@ class Logger {
     std::array<char, MaxMessageLength> mMessage{};
   };
 
-  moodycamel::ReaderWriterQueue<InternalLogData> mQueue{MaxNumMessages};
+  class InternalQueue {
+  public:
+    virtual ~InternalQueue() = default;
+    virtual bool tryEnqueue(InternalLogData &&value) = 0;
+    virtual bool tryDequeue(InternalLogData &value) = 0;
+  };
+  class InternalQueueSPSC : public InternalQueue {
+    moodycamel::ReaderWriterQueue<InternalLogData> mQueue{MaxNumMessages};
+
+  public:
+    bool tryEnqueue(InternalLogData &&value) override {
+      return mQueue.try_enqueue(std::move(value));
+    }
+    bool tryDequeue(InternalLogData &value) override {
+      return mQueue.try_dequeue(value);
+    }
+  };
+  class InternalQueueMPSC : public InternalQueue {
+    farbot::fifo<InternalLogData, farbot::fifo_options::concurrency::single,
+                 farbot::fifo_options::concurrency::multiple>
+        mQueue{MaxNumMessages};
+
+  public:
+    bool tryEnqueue(InternalLogData &&value) override {
+      return mQueue.push(std::move(value));
+    }
+    bool tryDequeue(InternalLogData &value) override {
+      return mQueue.pop(value);
+    }
+  };
+
+  std::unique_ptr<InternalQueue> mQueue{
+      Concurrency == QueueConcurrency::Single_Producer_Single_Consumer
+          ? (std::unique_ptr<InternalQueue>)
+                std::make_unique<InternalQueueSPSC>()
+          : std::make_unique<InternalQueueMPSC>()};
```

Comment:
nit: 

perhaps a case statement would be more clear and more extensible for the future

Reply:
agreed!